### PR TITLE
Bug fixes from demo walkthrough

### PR DESCRIPTION
**Pull request by auxesis**
- Make URL handling work (kept appending paths onto URLs instead of setting them new every time)
- Add some more informative debugging messages when running the `deploy` subcommand
- Refactor filelist generation, add tests, and make it more robust

### CODE REVIEW

**Comment by auxesis** [comment]
Test is failing but only intermittently, and only in GitHub Actions 😦 . 

Should merge cleanly

File: commands/deploy.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ const MaxFileSize = 1073741824 // 1GB
 // DeployCmd handles deploying an app to Section.
 type DeployCmd struct {
 	Debug     bool
+	Directory string `default:"./"`
```
Comment (pavelnikolov) [inline commands/deploy.go]: Maybe this could work just with a `.`.

File: commands/deploy_test.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+func TestCommandsDeployBuildFilelistErrorsOnNonExistentDirectory(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup
+	testCases := []string{
+		filepath.Join("testdata", "deploy", "non-existent-tree"),
+		filepath.Join("testdata", "deploy", "file"),
+	}
+	ignores := []string{}
```
Comment (pavelnikolov) [inline commands/deploy_test.go]: (nitpick) I prefer the default values:
`var ignores []string`

**Review by pavelnikolov — APPROVED** [review]

File: commands/deploy_test.go (hunk 2 above)
Comment (auxesis) [reply commands/deploy_test.go]: Good pickup, I prefer that syntax too. 

Fixed in 9a32807.

File: commands/deploy.go (hunk 1 above)
Comment (auxesis) [reply commands/deploy.go]: Yep good call. 

I did some testing, and this works fine. 

Changed in 871ba3c55c15bf5e0cc0db93d6ee633335d1fe8c.

**Pull request merged**